### PR TITLE
Review/add the API doc (`geo` folder - part 1)

### PR DESCRIPTION
**Pull request by HalynaDumych**
In  olp-cpp-sdk-core/include/olp/core/geo/coordinates&projection,
review and modify the existing documentation and add missing
descriptions so that we have fewer errors when building the API ref.

Relates-To: OLPEDGE-1444
Signed-off-by: Halyna Dumych <ext-halyna.dumych@here.com>

### CODE REVIEW

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h [hunk 1]
```diff
@@ -27,174 +27,175 @@ namespace olp {
...
+ * east and -180 degrees west.
  *
  * Internal representation of angles is radians.
  */
 class CORE_API GeoCoordinates {
  public:
-  /**
-   * @brief Construct default invalid geo coordinates.
-   */
+  ///@brief Creates a `GeoCoordinates` instance with invalid coordinates.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h:40]:
```suggestion
  /// Creates a `GeoCoordinates` instance with invalid coordinates.
```

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h [hunk 2]
```diff
@@ -27,174 +27,175 @@ namespace olp {
...
 
   /**
-   * @brief Construct geo coordinates form latitude and longitude.
+   * @brief Creates a `GeoCoordinates` instance from latitude and longitude.
    *
-   * @param latitude_radians WGS84 latitude in radians. Valid values are in
-   * [-Pi/2, Pi/2] range.
-   * @param longitude_radians WGS84 longitude in radians. Valid values are in
-   * [-Pi, Pi) range. Use normalized() to put geo coordinate in a valid range.
+   * @note Use `normalized()` to put a coordinate in a valid range.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h:46]:
```suggestion
   * @note Use `Normalized()` to put a coordinate in a valid range.
```

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h [hunk 3]
```diff
@@ -27,174 +27,175 @@ namespace olp {
...
+   * @param longitude_radians The WGS84 longitude in radians. Valid values are in
+   * the [-pi, pi] range.
    */
   GeoCoordinates(double latitude_radians, double longitude_radians);
 
   /**
-   * @brief Construct geo coordinates form latitude and longitude in degrees.
+   * @brief Creates a `GeoCoordinates` instance from latitude and longitude.
+   *
+   * @note Use `normalized()` to put a coordinate in a valid range.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h:58]: Same.

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h [hunk 4]
```diff
@@ -27,174 +27,175 @@ namespace olp {
...
 
   /**
-   * @brief Create GeoCoordinates from latitude and longitude in degrees.
+   * @brief Creates a `GeoCoordinates` instance from latitude and longitude.
    *
-   * @param latitude_degrees WGS84 latitude in degrees. Valid value is in [-90,
-   * 90] range.
-   * @param longitude_degrees WGS84 longitude in degrees. Valid value is in
-   * [-180, 180) range.
+   * @note Use normalized() to put a value in a valid range.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoCoordinates.h:72]: Same.

File: olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoRectangle.h [hunk 5]
```diff
@@ -26,48 +26,138 @@ namespace olp {
...
+ * @see `GeoBox` to add altitude.
  */
 class CORE_API GeoRectangle {
  public:
-  /** Constructs an empty rectangle. */
+  /// Creates an empty rectangle.
   GeoRectangle();
 
+  /**
+   * @brief Creates a `GeoRectangle` insatnce.
```
Comment (andescu) [inline olp-cpp-sdk-core/include/olp/core/geo/coordinates/GeoRectangle.h:39]:
```suggestion
   * @brief Creates a `GeoRectangle` instance.
```

**Review by andescu — APPROVED** [review]

**Review by mykhailo-kuchma — APPROVED** [review]

**Pull request merged**